atlp-rwanda/atlp-devpulse-fn
# General search and filtration admin

### PR Description
This PR implements functionalities for searching and filtering records across admin tables. These updates allow admin to efficiently see all data by searching for specific records and applying filters to enhance the user experience within the admin dashboard.

### Description of tasks that were expected to be completed
- Search Bar Integration
- Enabled filtering support
- Optimized the search and filtration functionalities for handling large datasets

### How to test this PR(locally)
Clone the repo and check the branch  `ft-general-search-admin`.
Set up environment variables as indicated in .env.example.
Run npm install followed by npm run dev to start the application.
 - **Email:** `admin@example.com`
   - **Password:** `password123` 
Check that searching and filtering functions work as expected

Deployed Link


Track PR (Issue number & link)
N/A

Screenshots (If appropriate)
<img width="959" alt="Screenshot 2024-10-08 112528" src="https://github.com/user-attachments/assets/b4351823-3472-4d31-86d4-86bf73fc3cdc">
<img width="959" alt="Screenshot 2024-10-08 112627" src="https://github.com/user-attachments/assets/7f00e970-1c8d-46ad-b972-f67a85f171ac">
<img width="959" alt="Screenshot 2024-10-08 112720" src="https://github.com/user-attachments/assets/56a19bf0-2494-46b8-a9be-2bf3ad73643d">
<img width="959" alt="Screenshot 2024-10-08 112811" src="https://github.com/user-attachments/assets/b01e6ca3-abe5-4f7a-932c-c5740eab6fd3">
